frontend/lib/ui/page_filters/page_filters.dart

```diff
@@ -50,16 +53,33 @@ class PageFiltersView extends ConsumerWidget {
             );
           }
 
+          int filterGroupIndex = 0;
+          int filtersCounted = filterGroups[filterGroupIndex].filters.length;
+          while (i > filtersCounted) {
```

Comment:
Just to make sure I understand this part correctly, the `while` loop here is to keep the count of the filter index, because now we store a list of filters, rather than one individual filter.`

Reply:
Yes, specifically it's to figure out which filter group this filter belongs to. The idea is that each filter group filters different types objects.